plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -61,6 +61,9 @@ type CiscoTelemetryMDT struct {
 	mutex           sync.Mutex
 	acc             telegraf.Accumulator
 	wg              sync.WaitGroup
+
+	// Required embed to maintain compatibility
+	dialout.UnimplementedGRPCMdtDialoutServer
```

Comment:
Can you elaborate a bit on _why_ it is needed? It was not there previously and I don't see how it is used in the code...

Reply:
`protoc-gen-go-grpc` requires that this be embedded in any struct that uses the proto-gRPC generated types, as it is meant to provide easy forward-compatibility if the generator changes, so that implemented server code doesn't break.

More realistically though, there is an inaccessible member function `mustEmbedUnimplementedGRPCMdtDialoutServer()` created as part of the generated code, that without this embed in the struct, calling `dialout.RegisterGRPCMdtDialoutServer()` on line 171 would give an error. Since it is inaccessible, embedding this is the best way to resolve the issue, and is recommended in a generated code comment (from `dialout_grpc.pb.go#71-73`):
```
// UnimplementedGRPCMdtDialoutServer must be embedded to have forward compatible implementations.
type UnimplementedGRPCMdtDialoutServer struct {
}
```

Reply:
Understood. Can you please summarize this as a comment in the code as otherwise someone will remove this and stuff starts to break. :-)

Reply:
Can you please add a comment here?
```suggestion
	// Though unused in the code, required by protoc-gen-go-rpc to maintain (forward-)compatibility
	dialout.UnimplementedGRPCMdtDialoutServer
```